### PR TITLE
Implement some new operations for elliptic curve

### PR DESCRIPTION
**Pull request by tcoratger**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behavior?

Add some missing operations for elliptic curve points.

## What is the new behavior?

- Sub operations
- Neg operations
- is_identity check

## Does this introduce a breaking change?

No

### CODE REVIEW

File: crates/starknet-types-core/src/curve/affine_point.rs [hunk 1]
```diff
@@ -34,6 +38,15 @@ impl AffinePoint {
     }
 }
 
+impl core::ops::Neg for &AffinePoint {
+    type Output = AffinePoint;
+
+    fn neg(self) -> AffinePoint {
+        // AffinePoint(ShortWeierstrassProjectivePoint::from_affine(self.x().0, -self.y().0).unwrap())
```
Comment (LucasLvy) [inline crates/starknet-types-core/src/curve/affine_point.rs:45]: what's that ?

File: crates/starknet-types-core/src/curve/projective_point.rs [hunk 2]
```diff
@@ -75,6 +79,28 @@ impl ops::AddAssign<&ProjectivePoint> for ProjectivePoint {
...
+    }
+}
+
+impl core::ops::Sub<&ProjectivePoint> for &ProjectivePoint {
+    type Output = ProjectivePoint;
+
+    fn sub(self, rhs: &ProjectivePoint) -> ProjectivePoint {
+        self + &-rhs
+    }
+}
```
Comment (LucasLvy) [inline crates/starknet-types-core/src/curve/projective_point.rs:96]: can you also implement `for ProjectivePoint` plz

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: crates/starknet-types-core/src/curve/affine_point.rs (hunk 1 above)
Comment (tcoratger) [reply crates/starknet-types-core/src/curve/affine_point.rs]: removed

File: crates/starknet-types-core/src/curve/projective_point.rs (hunk 2 above)
Comment (tcoratger) [reply crates/starknet-types-core/src/curve/projective_point.rs:110]: Done, did the same also for the addition.

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**